### PR TITLE
Add PyG remote backend example.

### PR DESCRIPTION
**Pull request by coledie**
- [x] Forked repo is synced with upstream -> github shows no code delta outside of the desired.
    https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork
- [x] Tests are passing? https://github.com/microsoft/DeepGNN/blob/main/CONTRIBUTING.md#run-tests
- [x] Changelog and documentation updated.
- [x] PR is labeled using the label menu on the right side.

New Behavior
----------------
Add example with PyG remote backend example.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -127,3 +127,5 @@ jobs:
         run: cd ./tensorflow/han/ && sudo chmod +x *.sh && bash ./run.sh CPU cleanup
       - name: Run gat examples
         run: cd ./tensorflow/gat/ && sudo chmod +x *.sh && bash ./run.sh CPU cleanup
+      - name: Run PyG example
+        run: cd ./pytorch/pyg/ && sudo chmod +x *.py && python main.py
```
Comment (mortid0) [inline .github/workflows/ci.yml:131]: all torch examples are run with tests now:
https://github.com/microsoft/DeepGNN/blob/a57ca28fe81d68196af57a4f4a9559f453febd44/.github/actions/test/action.yml#L34

File: examples/pytorch/pyg_interface.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
+"""PyGeo interface training example."""
```
Comment (mortid0) [inline examples/pytorch/pyg_interface.py:1]: forgot copyright

File: examples/pytorch/pyg_interface.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    for data in tqdm.tqdm(loader):
+        data = data.to(device)
+        data = data[("0", "0", "0")]
+        optimizer.zero_grad()
+
+        h = model(data.x, data.edge_index)
+
+        h_src = h[data.edge_label_index[0]]
+        h_dst = h[data.edge_label_index[1]]
+        link_pred = (h_src * h_dst).sum(dim=-1)  # Inner product.
```
Comment (mortid0) [inline examples/pytorch/pyg_interface.py:120]: Please, avoid [copy pasting code](https://github.com/pyg-team/pytorch_geometric/blob/02cc18d9d6841ecda4eb61eebb48b86f1aaae477/examples/graph_sage_unsup_ppi.py#L52) from other repos even with source comments

Comment (coledie) [reply examples/pytorch/pyg_interface.py:120]: The problem is this interface is meant to be a remote backend for PyG so I need to use their model and dataloader and just copy the glue code from an example since it is not importable. What should I do instead?

Comment (mortid0) [reply examples/pytorch/pyg_interface.py:120]: Write train loop on your own that is similar to theirs.

**Review by mortid0 — APPROVED** [review]
